src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverterTests.HandleNull.cs

```diff
@@ -0,0 +1,484 @@
+                {
+                    return -1;
+                }
+
+                throw new JsonException();
+            }
+
+            public override void Write(Utf8JsonWriter writer, int? value, JsonSerializerOptions options)
+            {
+                if (!value.HasValue)
+                {
+                    writer.WriteNumberValue(-1);
+                    return;
+                }
+
+                throw new NotSupportedException();
+            }
+        }
+
+        [Fact]
+        public static void NullableValueTypeConverter_OptOut()
+        {
+            // Baseline
+            int? val = JsonSerializer.Deserialize<int?>("null");
+            Assert.Null(val);
+            Assert.Equal("null", JsonSerializer.Serialize(val));
+
+            // Per null handling opt-out, serializer handles null.
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new NullableInt32NullConverter_NullOptOut());
+
+            val = JsonSerializer.Deserialize<int?>("null", options);
+            Assert.Null(val);
+            Assert.Equal("null", JsonSerializer.Serialize(val, options));
+        }
+
+        private class NullableInt32NullConverter_NullOptOut : NullableInt32NullConverter_SpecialCaseNull
+        {
+            public override bool HandleNull => false;
+        }
+
+        [Fact]
+        public static void ReferenceTypeConverter_NoOverride()
+        {
+            // Baseline
+            Uri val = JsonSerializer.Deserialize<Uri>("null");
+            Assert.Null(val);
+            Assert.Equal("null", JsonSerializer.Serialize(val));
+
+            // Per null handling default value for reference types (false), serializer handles null.
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new Int32NullConverter_SpecialCaseNull());
+
+            val = JsonSerializer.Deserialize<Uri>("null", options);
+            Assert.Null(val);
+            Assert.Equal("null", JsonSerializer.Serialize(val, options));
+        }
+
+        private class UriNullConverter_SpecialCaseNull : JsonConverter<Uri>
+        {
+            public override Uri Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType == JsonTokenType.Null)
+                {
+                    return new Uri("https://default");
+                }
+
+                throw new JsonException();
+            }
+
+            public override void Write(Utf8JsonWriter writer, Uri value, JsonSerializerOptions options)
+            {
+                if (value == null)
+                {
+                    writer.WriteStringValue("https://default");
+                    return;
+                }
+
+                throw new NotSupportedException();
+            }
+        }
+
+        [Fact]
+        public static void ReferenceTypeConverter_OptOut()
```

Comment:
Not used. Also suggest to have "Uri" in the name.

Reply:
This is a test method, so there no references to this. I wanted to have the generic because the behavior is them same for all reference types, not just Uri.